src/common/downloadFileDialog.component.tsx

```
@@ -0,0 +1,127 @@
+import { AxiosError } from 'axios';
+import React from 'react';
+import { APIImage, APIImageWithURL } from '../api/api.types';
+import handleIMS_APIError from '../handleIMS_APIError';
+import { downloadFileByLink } from '../utils';
+
+export interface BaseDownloadFileProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDownloadDialogProps extends BaseDownloadFileProps {
+  fileType: 'Image';
+  getFile: (id: string) => Promise<APIImageWithURL>;
+  file: APIImage;
+}
+
+export type DownloadFileProps = ImageDownloadDialogProps;
+
+const DownloadFileDialog = (props: DownloadFileProps) => {
+  const { open, onClose, fileType, file, getFile } = props;
+
+  const [formError, setFormError] = React.useState<string | undefined>(
+    undefined
+  );
+
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const queryClient = useQueryClient();
+
+  const handleClose = React.useCallback(() => {
+    queryClient.removeQueries({
+      queryKey: ['Image', file.id],
+    });
+    setFormError(undefined);
+    setIsLoading(false);
+    onClose();
+  }, [onClose, file, queryClient]);
+
+  const handleDownloadFile = React.useCallback(
+    (data: APIImageWithURL | undefined) => {
+      if (data) {
+        downloadFileByLink(document, data.download_url, data.file_name);
+        onClose();
+      } else {
+        setFormError('No data provided. Please refresh and try again');
+      }
+    },
+    [onClose]
+  );
+
+  const handleClick = React.useCallback(async () => {
+    setIsLoading(true);
+
+    let data: APIImageWithURL | undefined = undefined;
+    let closing: boolean = false;
+    try {
+      const fetchedData = await queryClient.fetchQuery({
+        queryKey: ['Image', file.id],
+        queryFn: () => getFile(file.id),
+      });
+      data = fetchedData;
+    } catch (error) {
+      if (error instanceof CancelledError) closing = true;
```

Did you manage to trigger this error this when you manually testing it ? and if so how ??

---

This error gets triggered when the dialog is closed by `queryClient.removeQueries` **IF** the query is still fetching/refetching when you close it.

I tested it by trying to download the 5th image in any gallery using mock data (since in the handlers, the 5th image always returns an error).

---

I've changed the implementation so that it does not retry. I've since removed this part of the implementation